### ISSUE TITLE
Help

### ISSUE THREAD

**Issue by mhmmd10** [sandeepyadav1478/Uploading-video-on-youtube#3]
Hi
i want to run this
but iam beginner
you can make video . how to run and also add http://localhost:8080/ to Authorized redirect URIs in OAuth 2.0 Client IDs
i need help bro.

**Comment by sandeepyadav1478** [comment]
Hmm!
it will take more time to edit and upload on YouTube.
I think, U can follow these steps on google (It have basic UI, just read and follow):

1. Go here [Developers.google.com](https://developers.google.com/identity/protocols/oauth2#1.-obtain-oauth-2.0-credentials-from-the-dynamic_data.setvar.console_name-.)

2. Create credentials, you can follow this link also if 1st one fails [developers.google.com](https://developers.google.com/identity/sign-in/web/sign-in)

3. Go to 'Credentials' tab.

4. Click on **+ Create Credentials** button and **OAuth client ID**

5. Select **Web application**. I leave rest on you. You will find **Authorized redirect URIs** at bottom of this page, add that address there.

### Youtube api v3 credentials:
1. Click on **+ ENABLE APIS & SERVICES** on sidebar at same dashboard.

2. Scroll down to **Youtube data Api v3**, click on it. Then click on **MANAGE**.

3. Click on **Credentials** on sidebar. I think you can carry on from here:)

_If you still need any help feel free to contact me on this repo_

**Comment by mhmmd10** [comment]
/content/new.json-youtube-v3-(['https://www.googleapis.com/auth/youtube.upload'],)
['https://www.googleapis.com/auth/youtube.upload']
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
<ipython-input-10-861957089dcb> in <module>()
     57 
     58 
---> 59 service = Create_Service(CLIENT_SECRET_FILE, API_NAME, API_VERSION, SCOPES)
     60 
     61 upload_date_time = datetime.datetime(2021, 8, 7, 2, 3, 0).isoformat() + '.000Z' #

6 frames
/usr/lib/python3.7/socketserver.py in server_bind(self)
    464         if self.allow_reuse_address:
    465             self.socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
--> 466         self.socket.bind(self.server_address)
    467         self.server_address = self.socket.getsockname()
    468 

OSError: [Errno 98] Address already in use

**Comment by mhmmd10** [comment]
error

**Comment by sandeepyadav1478** [comment]
Which machine you are using to run this code?
Try Google Colab!
Code works on both local and Colab machine.
But I suggest to use colab coz it works on linux server, and max. developers usage linux as there base OS. So maybe those api works good on Colab.

**Comment by mhmmd10** [comment]
with google colab

**Comment by sandeepyadav1478** [comment]
yes. And if you need more help, Then create meeting link and send it to me.
